app/workers/cron_job.rb

```diff
@@ -1,30 +1,24 @@
 # frozen_string_literal: true
 
 module CronJob
-  class Worker
-    include Sidekiq::Worker
-
-    def perform(task)
-      Rails.logger.debug("Executing cron task #{task}")
-      instance_exec(task, &ThreeScale::Jobs::JOB_PROC)
-    end
-
-    def rake(task)
-      system('rake', task)
-    end
-
-    def runner(command)
-      system('rails', 'runner', command)
+  class Worker < ApplicationJob
+
+    def perform(hash)
```

Comment:
We're changing the API. We need to be careful while deploying this, to make sure there are no jobs currently enqueued.

It might be new to Saas, but on-prem customers could still be affected.

Reply:
I will create another class that only uses this one.
Thanks for the review

Reply:
In fact, it should be inside the documentation for the upgrade. What should be done is:

- disabling the sidekiq cron tasks
- when all sidekiq cron related tasks have been all processed make the upgrade

The issue why we can't have a smooth transition is that because of the Cron job enqueuer. So better stick to a documentation upgrade

Reply:
I would prefer the documentation route as well so we can avoid an extra class to maintain. Let's just make sure that works for deployments with the 3scale operator as well (not only for deployment with OpenShift templates).

cc @miguelsorianod @eguzki

Reply:
So last commit makes it backward compatible, when we do not need it anymore on future version we will remove that